dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/DependencyInjectionTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq.Expressions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.VectorData;
+using Xunit;
+
+namespace VectorDataSpecificationTests;
+
+public abstract class DependencyInjectionTests<TVectorStore, TCollection, TKey, TRecord>
+    where TVectorStore : VectorStore
+    where TCollection : VectorStoreCollection<TKey, TRecord>
+    where TKey : notnull
+    where TRecord : class
+{
+    protected abstract void PopulateConfiguration(ConfigurationManager configuration, object? serviceKey = null);
+
+    protected abstract void RegisterVectorStore(IServiceCollection services, ServiceLifetime lifetime, object? serviceKey = null);
+
+    protected abstract void RegisterCollection(IServiceCollection services, ServiceLifetime lifetime, string collectionName = "name", object? serviceKey = null);
```

Comment:
@roji regarding your comment about exposing a list of delegates from https://github.com/microsoft/semantic-kernel/pull/11594/files#r2049050430, I propose to take what I have now and let me shape the final form of the base DI test class in the next PR, once I apply the same pattern to all the connectors and have the full picture

Reply:
Absolutely no problem to finalize the tests later! I do think we'll ideally want to cover more than just one register method (which is what this allows), but definitely ok for now.